xtask/src/util.rs

```diff
@@ -278,12 +278,18 @@ pub fn project_dir(msg_info: &mut MessageInfo) -> cross::Result<PathBuf> {
 
 // note: for GHA actions we need to output these tags no matter the verbosity level
 pub fn gha_print(content: &str) {
-    println!("{}", content)
+    #[cfg(not(test))]
+    println!("{}", content);
+    #[cfg(test)]
+    eprintln!("{}", content);
 }
 
 // note: for GHA actions we need to output these tags no matter the verbosity level
 pub fn gha_error(content: &str) {
```

Comment:
I think this can be converted to a single helper macro.

```rust
diff --git a/xtask/src/util.rs b/xtask/src/util.rs
index 7c377ef..edac27e 100644
--- a/xtask/src/util.rs
+++ b/xtask/src/util.rs
@@ -276,20 +276,23 @@ pub fn project_dir(msg_info: &mut MessageInfo) -> cross::Result<PathBuf> {
     Ok(cargo_metadata(msg_info)?.workspace_root)
 }

+macro_rules! gha_output {
+    ($fmt:literal$(, $args:expr)* $(,)?) => {
+        #[cfg(not(test))]
+        println!($fmt $(, $args)*);
+        #[cfg(test)]
+        eprintln!($fmt $(,$args)*);
+    };
+}
+
 // note: for GHA actions we need to output these tags no matter the verbosity level
 pub fn gha_print(content: &str) {
-    #[cfg(not(test))]
-    println!("{}", content);
-    #[cfg(test)]
-    eprintln!("{}", content);
+    gha_output!("{}", content);
 }

 // note: for GHA actions we need to output these tags no matter the verbosity level
 pub fn gha_error(content: &str) {
-    #[cfg(not(test))]
-    println!("::error {}", content);
-    #[cfg(test)]
-    eprintln!("::error {}", content);
+    gha_output!("::error {}", content);
 }

 #[track_caller]
@@ -298,10 +301,7 @@ pub fn gha_output(tag: &str, content: &str) {
         // https://github.com/actions/toolkit/issues/403
         panic!("output `{tag}` contains newlines, consider serializing with json and deserializing in gha with fromJSON()")
     }
-    #[cfg(not(test))]
-    println!("::set-output name={tag}::{}", content);
-    #[cfg(test)]
-    eprintln!("::set-output name={tag}::{}", content);
+    gha_output!("::set-output name={tag}::{}", content);
 }

 pub fn read_dockerfiles(msg_info: &mut MessageInfo) -> cross::Result<Vec<(PathBuf, String)>> {
```